src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -594,7 +625,7 @@ public OpenIdConnectHandler(HttpClient backchannel, HtmlEncoder htmlEncoder)
                     // response, since it's not a valid parameter when using the implicit flow.
                     // See http://openid.net/specs/openid-connect-core-1_0.html#Authentication
                     // and https://tools.ietf.org/html/rfc6749#section-4.2.2.
-                    SaveTokens(ticket.Principal, message, saveRefreshToken: false);
+                    SaveTokens(ticket.Principal, message, jwt.Issuer, saveRefreshToken: false);
```

Comment:
I might be misunderstanding the matter here. My understanding is that the token WAS issued by the IDP, and if we are executing this code it means that we validated it as such. The fact that we obtained it via backchannel rather than front channel should be immaterial, *as long as we did validate the token and its origin as being the IDP*. Is there anything inaccurate in this?